comp/logs/agent/config/endpoints.go

```diff
@@ -238,6 +267,52 @@ func (e *Endpoint) GetStatus(prefix string, useHTTP bool) string {
 	return fmt.Sprintf("%sSending %s logs in %s to %s on port %d", prefix, compression, protocol, host, port)
 }
 
+// onConfigUpdate handles configuration change notification to update the internal API key of the Endpoint if needed
+func (e *Endpoint) onConfigUpdate(l *LogsConfigKeys) {
+	l.getConfig().OnUpdate(func(key string, oldVal interface{}, newVal interface{}) {
+		if key != e.configSettingPath {
+			return
+		}
+
+		// main Endpoints can directly get their API key from the configuration without having to load complex
+		// types.
+		if !e.isAdditionalEndpoint {
+			if newAPIKey, ok := newVal.(string); !ok {
+				log.Errorf("new API key for '%s' is invalid (not a string) ignoring new value", e.configSettingPath)
+			} else {
+				if oldKey, ok := oldVal.(string); ok && oldKey != e.apiKey.Load() {
+					// This should never happens as it means that an update from the config was
+					// missed
+					log.Warnf("old API key for '%s' doesn't match the one in this endpoints", e.configSettingPath)
+				}
+				log.Infof("rotating API key for '%s': %s -> %s",
+					e.configSettingPath,
+					scrubber.HideKeyExceptLastFiveChars(e.apiKey.Load()),
+					scrubber.HideKeyExceptLastFiveChars(newAPIKey),
+				)
+				e.apiKey.Store(newAPIKey)
+			}
+			return
+		}
+
+		newAdditionalEndpoints, _ := l.getAdditionalEndpoints()
+		if len(newAdditionalEndpoints) <= e.additionalEndpointsIdx {
```

Comment:
Minor nit: could this conditional be reversed? I'm much more used to seeing the pattern `index >= length` rather than the opposite.